### PR TITLE
Fix #69 Adapt URI

### LINKED ISSUE #58: Awkward API naming in ModelController::readResource(...)
As mentioned in a [comment](https://github.com/eclipsesource/modelserver/pull/57#discussion_r311063544) on PR #57, the naming of the `ModelController::readResource(...)` method makes for difficult code reading at call sites such as `ModelController::update(...)`.  This method may or may not implicitly create the resource being "read", which is not intuitive, and in some cases (as in `update`) is a side-effect that needs to be reverted.

### LINKED ISSUE #69: Make adaption of modeluri path more robust
See https://github.com/eclipsesource/modelserver/pull/63#discussion_r311379301 for context

### LINKED ISSUE #124: Let ModelRepository actually create resource
Currently, the `ModelRepository` is relying on the resource already being created by the codec, which is not the indented way of how the codec nor the model repository are expected to work.

### LINKED ISSUE #126: Commands are deserialized with relative URI
Commands coming in over the wire on the `edit` endpoint are deserialized in a temporary resource in the *Model Server*'s resource set.  This resource has a dummy URI that is a relative URI.  It needs to be an absolute URI in the scope of the workspace in order to properly resolve references from the command model into the user model, when those proxy HREFs have only the relative URI by which the client knows its model resource.

### PR DESCRIPTION
**Pull request by cdamus**
- make adaptUri() method more robust (e.g. for spaces in path)
- also fix #126 command deserialization with absolute resource URI
- remove decode overload which expected RS and Model URI
- refactor ModelRepository#addModel to not only get resource, but actually also create it (fixes #124)
- Rename and refactor `readResource` to `readPayload` (fixes #58 )

### CODE REVIEW

File: com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/codecs/DefaultJsonCodec.java [hunk 1]
```diff
@@ -53,7 +48,30 @@ public JsonNode encode(EObject obj) throws EncodingException {
...
+			jsonResource.getErrors().add(new JSONException(e, JsonLocation.NA));
+		}
+
+		return Optional.of(jsonResource.getContents().remove(0));
+	}
+	
+	@Override
+	public Optional<EObject> decode(URI workspaceURI, String payload) throws DecodingException {
+		URI uri = URI.createURI("virtual.json").resolve(workspaceURI);
+		final JsonResource jsonResource = new JsonResource(uri, getObjectMapper());
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/codecs/DefaultJsonCodec.java:61]: The body is the same as for the other overload, could we just extract a method here or unify both methods?

File: com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/codecs/XmiCodec.java [hunk 2]
```diff
@@ -53,7 +53,14 @@ public JsonNode encode(EObject eObject) throws EncodingException {
 	@Override
 	public Optional<EObject> decode(String payload) throws DecodingException {
 		ResourceSet resourceSet = new ResourceSetImpl();
-		Optional<Resource> resource = decode(resourceSet, "virtual.xmi", payload);
+		Optional<Resource> resource = decode(resourceSet, "virtual.xmi", null, payload);
+		return resource.map(r -> r.getContents().isEmpty() ? null : r.getContents().get(0));
+	}
+	
+	@Override
+	public Optional<EObject> decode(URI workspaceURI, String payload) throws DecodingException {
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.common/src/main/java/com/eclipsesource/modelserver/common/codecs/XmiCodec.java]: We could unify with this method with the one from above.

**Review by edgarmueller — APPROVED** [review]
Looks good to me!

**Comment by ndoschek** [comment]
Summary of fixed issues with this PR:
Fixes #58 
Fixes #69 
Fixes #95 
Fixes #124 
Fixes #126

**Pull request merged**